src/compiler/diagnosticMessages.json

```diff
@@ -5235,6 +5235,14 @@
         "category": "Message",
         "code": 6276
     },
+    "Resolution of non-relative name failed; trying with modern Node resolution features disabled to see if npm library needs configuration update.": {
+        "category": "Message",
+        "code": 6277
+    },
+    "There are types at '{0}', but this result could not be resolved when respecting package.json \"exports\". The '{1}' library may need to update its package.json.": {

```

Comment:
Q: Would a similar error for `"imports"` for resolving `#`-prefixed nonrelative names be useful?